hazelcast/src/main/java/com/hazelcast/map/impl/SizeEstimator.java

```diff
@@ -16,18 +16,101 @@
 
 package com.hazelcast.map.impl;
 
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.nio.UnsafeHelper;
+import sun.misc.Unsafe;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeDataSupport;
+import java.lang.management.ManagementFactory;
+
 /**
  * Size estimator general contract.
  *
  * @param <T> the type of object which's size going to be estimated.
  */
-public interface SizeEstimator<T> {
+public abstract class SizeEstimator<T> {
```

Comment:
You can try this here, but I don't know if this is the correct level to put this annotation: `@SuppressWarnings("checkstyle:illegalimport")`